### PR TITLE
feat(@trezor/suite-web-landing): remove next.js & use custom build pipeline

### LINKED ISSUE #4115: Get rid of next on landing page
replace next.js builds in `@trezor/suite-web-landing` with  `@trezor/suite-build`

### PR DESCRIPTION
**Pull request by dimasivan**
fix #4115

### CODE REVIEW

File: packages/suite-build/configs/landing.webpack.config.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            patterns: [
+                'browser-detection',
+                'connect',
+                'fonts',
+                'guide',
+                'images',
+                'message-system',
+                'oauth',
+                'translations',
+                'videos',
```
Comment (szymonlesisz) [inline packages/suite-build/configs/landing.webpack.config.ts]: verify assets, i dont think all of them are required by landing page

File: packages/suite-web-landing/package.json [hunk 2]
```diff
@@ -7,28 +7,29 @@
...
     },
     "dependencies": {
         "@trezor/components": "1.0.0",
-        "next": "^10.0.5",
-        "polished": "^4.0.5",
-        "react": "17.0.1",
-        "react-awesome-reveal": "^3.5.2",
-        "react-dom": "17.0.1",
-        "react-intl": "^5.10.19"
+        "@trezor/suite": "21.9.0",
```
Comment (szymonlesisz) [inline packages/suite-web-landing/package.json]: @trezor/suite, redux and related should not be dependency of landing-page

File: packages/suite/src/components/suite/Labeling/components/Metadata/index.tsx [hunk 3]
```diff
@@ -260,18 +260,16 @@ const MetadataLabeling = (props: Props) => {
     const onSubmit = async (value: string | undefined | null) => {
         isSubscribedToSubmitResult.current = props.payload.defaultValue;
         setPending(true);
-        const result = await addMetadata({
+        await addMetadata({
```
Comment (szymonlesisz) [inline packages/suite/src/components/suite/Labeling/components/Metadata/index.tsx]: suite code should not be affected

**Review by szymonlesisz — CHANGES_REQUESTED** [review]

File: packages/suite-build/configs/landing.webpack.config.ts (hunk 1 above)
Comment (dimasivan) [reply packages/suite-build/configs/landing.webpack.config.ts]: I tried to build Suite when removing plugins one by one, seems like only these four are required  68c7a59

File: packages/suite-web-landing/package.json (hunk 2 above)
Comment (dimasivan) [reply packages/suite-web-landing/package.json]: 68c7a59

File: packages/suite/src/components/suite/Labeling/components/Metadata/index.tsx (hunk 3 above)
Comment (dimasivan) [reply packages/suite/src/components/suite/Labeling/components/Metadata/index.tsx]: changes was made due to unnecessary packages, reverted in 68c7a59

File: packages/suite-web-landing/package.json [hunk 4]
```diff
@@ -7,28 +7,25 @@
...
         "@types/react-dom": "^17.0.0",
+        "@types/react-helmet": "^6.1.0",
         "@types/react-redux": "^7.1.7",
-        "babel-plugin-styled-components": "1.12.0",
-        "next-optimized-images": "^2.6.2",
-        "next-transpile-modules": "^4.1.0",
-        "next-videos": "^1.4.0",
-        "ts-loader": "^8.0.14",
-        "webpack": "^4.44.2"
+        "is-website-vulnerable": "^1.14.6",
```
Comment (matejkriz) [inline packages/suite-web-landing/package.json]: Is this a new dependency? Is it required?

File: packages/suite-web-landing/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (matejkriz) [inline packages/suite-web-landing/src/index.ts]: browser detection is not applied on current landing page and I believe it's correct

File: packages/suite-build/configs/landing.webpack.config.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+const baseDir = getPathForProject('landing');
+const config: webpack.Configuration = {
+    entry: [path.join(baseDir, 'src', 'index.ts')],
+    output: {
+        path: path.join(baseDir, 'build'),
+    },
+    plugins: [
+        new CopyPlugin({
+            patterns: ['browser-detection', 'fonts', 'images', 'translations'].map(dir => ({
```
Comment (matejkriz) [inline packages/suite-build/configs/landing.webpack.config.ts]: browser-detection is not applied on current landing page

File: packages/suite-web-landing/src/static/index.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+            media="all"
+            rel="stylesheet"
+            href="<%= assetPrefix %>/static/fonts/fonts.css"
+        />
+        <link rel="icon" href="<%= assetPrefix %>/static/images/icons/favicon/favicon.png" />
+        <link
+            rel="apple-touch-icon"
+            href="<%= assetPrefix %>/static/images/icons/favicon/favicon.png"
+        />
+        <script src="<%= assetPrefix %>/static/browser-detection/index.js"></script>
```
Comment (matejkriz) [inline packages/suite-web-landing/src/static/index.html]: browser-detection is redundant

File: packages/suite-web-landing/src/static/index.html [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<html lang="en" translate="no">
+    <head>
```
Comment (matejkriz) [inline packages/suite-web-landing/src/static/index.html:3]: please compare the head section on this branch with the current one (and this file with original deleted _document.tsx)

Comment (matejkriz) [reply packages/suite-web-landing/src/static/index.html:3]: for example onion-location and cache directives are missing

File: packages/suite-web-landing/pages/start.tsx [hunk 9]
```diff
@@ -1,137 +0,0 @@
-import React, { useState } from 'react';
```
Comment (matejkriz) [inline packages/suite-web-landing/pages/start.tsx:1]: We still need /start page, it's a landing page for every new owner of Trezor device... It looks similar to index, but translation keys (and texts) are different.

File: packages/suite-build/configs/landing.webpack.config.ts [hunk 10]
```diff
@@ -0,0 +1,53 @@
+import path from 'path';
+import webpack from 'webpack';
+import CopyPlugin from 'copy-webpack-plugin';
+import HtmlWebpackPlugin from 'html-webpack-plugin';
+import ReactRefreshWebpackPlugin from '@pmmmwh/react-refresh-webpack-plugin';
+
+import routes from '../../suite/src/config/suite/routes';
```
Comment (matejkriz) [inline packages/suite-build/configs/landing.webpack.config.ts]: those are suite routes not landing page routes (landing page has only two routes so far `/` and `/start`)

File: packages/suite-web-landing/src/static/index.html (hunk 7 above)
Comment (dimasivan) [reply packages/suite-web-landing/src/static/index.html]: 14a42c2

File: packages/suite-web-landing/src/static/index.html (hunk 8 above)
Comment (dimasivan) [reply packages/suite-web-landing/src/static/index.html:3]: 14a42c2

File: packages/suite-web-landing/pages/start.tsx (hunk 9 above)
Comment (dimasivan) [reply packages/suite-web-landing/pages/start.tsx:1]: 14a42c2

File: packages/suite-build/configs/landing.webpack.config.ts (hunk 10 above)
Comment (dimasivan) [reply packages/suite-build/configs/landing.webpack.config.ts]: 14a42c2

File: packages/suite-web/src/static/index.html [hunk 11]
```diff
@@ -3,6 +3,9 @@
     <head>
         <meta name="google" content="notranslate" />
         <meta charset="utf-8">
+        <meta httpEquiv="Pragma" content="no-cache" />
+        <meta httpEquiv="cache-control" content="no-cache" />
+        <meta httpEquiv="expires" content="-1" />
```
Comment (tomasklim) [inline packages/suite-web/src/static/index.html]: What is the reason for adding cache directives to suite-web here?  If you wanted to add it to suite-web-landing, then all directives make sense.

For suite-web I think that we do not need `<meta httpEquiv="Pragma" content="no-cache" />` as we support only Chrome and Firefox. Not sure about `<meta httpEquiv="expires" content="-1" />`.

File: packages/suite-web-landing/src/Main.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import AppRouter from './support/Router';
+import { store } from '@suite/reducers/store';
+import history from '@suite/support/history';
+
+const Main = () => {
+    const [translationMode, setTranslationMode] = useState(false);
+
+    return (
+        <>
+            <ReduxProvider store={store}>
```
Comment (tomasklim) [inline packages/suite-web-landing/src/Main.tsx:21]: Do we need/use redux here?

File: packages/suite-web-landing/src/Main.tsx [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+                <ThemeProvider>
+                    <RouterProvider history={history}>
+                        <GlobalStyleProvider />
```
Comment (tomasklim) [inline packages/suite-web-landing/src/Main.tsx]: This component adds not only the old global styles but also `animations` and `notifications` styles which we do not need here. I would add some conditions there.

File: packages/suite-web-landing/package.json [hunk 14]
```diff
@@ -7,28 +7,24 @@
         "type-check:watch": "yarn type-check -- --watch",
         "lint": "eslint . --ext .tsx,.ts",
         "copy-files": "yarn workspace @trezor/suite-data copy-static-files",
-        "dev": "yarn workspace @trezor/suite-data copy-static-files && next",
-        "build:app": "next build && next export -o build",
-        "build": "run-s copy-files build:app"
+        "dev": "rimraf ./build && yarn workspace @trezor/suite-build run dev:landing",
+        "build": "rimraf ./build && yarn workspace @trezor/suite-build run build:landing"
     },
     "dependencies": {
```
Comment (tomasklim) [inline packages/suite-web-landing/package.json:13]: There are dependencies missing and there are some extras. Both in dependencies and devDependencies

File: packages/suite-build/webpack.config.ts [hunk 15]
```diff
@@ -8,6 +8,7 @@ import base from './configs/base.webpack.config';
 import dev from './configs/dev.webpack.config';
 import web from './configs/web.webpack.config';
 import desktop from './configs/desktop.webpack.config';
+import landing from './configs/landing.webpack.config';
 
 const configs = [base];
```
Comment (tomasklim) [inline packages/suite-build/webpack.config.ts:13]: Beware, that base config uses `browserslist` with
```
Chrome >= 59
Firefox >= 54
```
Landing page should be supported in most browsers. https://browserslist.dev/

File: packages/suite-build/utils/constants.ts [hunk 16]
```diff
@@ -1,7 +1,8 @@
-export const PROJECTS = ['web', 'desktop'] as const;
+export const PROJECTS = ['web', 'desktop', 'landing'] as const;
 export type Project = typeof PROJECTS[number];
 
 export const DEV_PORTS: { [key in Project]: number } = {
     web: 8000,
     desktop: 8000,
+    landing: 3000,
```
Comment (tomasklim) [inline packages/suite-build/utils/constants.ts:7]: Was there some discussion why it should run on a different port?

When I click on the `Trezor suite for web` button, it does not redirect to `localhost:8000`, so it does not make sense for me.

File: packages/suite-web-landing/src/support/Router.tsx [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+            {routes.map(route => (
+                <Route
+                    key={route.name}
+                    path={process.env.ASSET_PREFIX + route.pattern}
+                    exact={route.exact}
+                    component={components[route.name]}
+                />
+            ))}
+        </Switch>
+    </Suspense>
```
Comment (tomasklim) [inline packages/suite-web-landing/src/support/Router.tsx:36]: I think that we do not need to use `Suspense` and `lazy` here as there is no connection between those 2 pages

File: packages/suite-web-landing/components/Index/index.tsx [hunk 18]
```diff
@@ -0,0 +1,116 @@
+import React from 'react';
```
Comment (tomasklim) [inline packages/suite-web-landing/components/Index/index.tsx:1]: Let's discuss if we need to have both `Index` and `Start` components in different files. They are almost the same.

File: packages/suite-web-landing/package.json (hunk 14 above)
Comment (tomasklim) [reply packages/suite-web-landing/package.json:13]: Let's discuss if we need `polished` and `react-awesome-reveal` libs.

**Review by tomasklim — CHANGES_REQUESTED** [review]

File: ci/build.yml [hunk 19]
```diff
@@ -73,9 +73,10 @@ suite-web build stable codesign:
 suite-web-landing build dev:
   stage: build
   script:
+    - apt install chromium
```
Comment (vdovhanych) [inline ci/build.yml]: We shouldn't do it like this, if you need chromium to be a part of the pipeline for this, let's add it to our base image so we're not pulling it all the time.

**Review by vdovhanych — CHANGES_REQUESTED** [review]

**Pull request closed without merging**